internal/bundle/deploy_test.go

```diff
@@ -0,0 +1,125 @@
+		destroyBundle(t, ctx, bundleRoot)
+	})
+
+	// Assert the schema is created
+	catalogName := "main"
+	schemaName := "test-schema-" + uniqueId
+	schema, err := w.Schemas.GetByFullName(ctx, strings.Join([]string{catalogName, schemaName}, "."))
+	require.NoError(t, err)
+	require.Equal(t, strings.Join([]string{catalogName, schemaName}, "."), schema.FullName)
+	require.Equal(t, "This schema was created from DABs", schema.Comment)
+
+	// Assert the pipeline is created
+	pipelineName := "test-pipeline-" + uniqueId
+	pipeline, err := w.Pipelines.GetByName(ctx, pipelineName)
+	require.NoError(t, err)
+	require.Equal(t, pipelineName, pipeline.Name)
+	id := pipeline.PipelineId
+
+	// Assert the pipeline uses the schema
+	i, err := w.Pipelines.GetByPipelineId(ctx, id)
+	require.NoError(t, err)
+	require.Equal(t, catalogName, i.Spec.Catalog)
+	require.Equal(t, strings.Join([]string{catalogName, schemaName}, "."), i.Spec.Target)
+
+	// Create a volume in the schema, and add a file to it. This ensure that the
+	// schema as some data in it and deletion will fail unless the generated
+	// terraform configuration has force_destroy set to true.
+	volumeName := "test-volume-" + uniqueId
+	volume, err := w.Volumes.Create(ctx, catalog.CreateVolumeRequestContent{
+		CatalogName: catalogName,
+		SchemaName:  schemaName,
+		Name:        volumeName,
+		VolumeType:  catalog.VolumeTypeManaged,
+	})
+	require.NoError(t, err)
+	require.Equal(t, volume.Name, volumeName)
+
+	fileName := "test-file-" + uniqueId
+	err = w.Files.Upload(ctx, files.UploadRequest{
+		Contents: io.NopCloser(strings.NewReader("Hello, world!")),
+		FilePath: fmt.Sprintf("/Volumes/%s/%s/%s/%s", catalogName, schemaName, volumeName, fileName),
+	})
+	require.NoError(t, err)
+
+	return bundleRoot
+}
+
+func TestAccBundleDeployUcSchema(t *testing.T) {
+	ctx, wt := acc.UcWorkspaceTest(t)
+	w := wt.W
+
+	uniqueId := uuid.New().String()
+	schemaName := "test-schema-" + uniqueId
+	catalogName := "main"
+
+	bundleRoot := setupUcSchemaBundle(t, ctx, w, uniqueId)
+
+	// Remove the UC schema from the resource configuration.
+	err := os.Remove(filepath.Join(bundleRoot, "schema.yml"))
+	require.NoError(t, err)
+
+	// Redeploy the bundle
+	err = deployBundle(t, ctx, bundleRoot)
+	require.NoError(t, err)
+
+	// Assert the schema is deleted
+	_, err = w.Schemas.GetByFullName(ctx, strings.Join([]string{catalogName, schemaName}, "."))
+	apiErr := &apierr.APIError{}
+	assert.True(t, errors.As(err, &apiErr))
+	assert.Equal(t, "SCHEMA_DOES_NOT_EXIST", apiErr.ErrorCode)
+}
+
+func TestAccBundleDeployUcSchemaFailsWithoutAutoApprove(t *testing.T) {
+	ctx, wt := acc.UcWorkspaceTest(t)
+	w := wt.W
+
+	uniqueId := uuid.New().String()
+	bundleRoot := setupUcSchemaBundle(t, ctx, w, uniqueId)
+
+	// Remove the UC schema from the resource configuration.
+	err := os.Remove(filepath.Join(bundleRoot, "schema.yml"))
+	require.NoError(t, err)
+
+	// Redeploy the bundle
+	t.Setenv("BUNDLE_ROOT", bundleRoot)
+	t.Setenv("TERM", "dumb")
```

Comment:
This line prevents the `cmdio` from registering the test run as from an interactive terminal. Ideally this should not be required but I'm not sure how to fix it in upstream code ATM. Taking a note to look at this as a followup, since determining terminal type has been an underinvested part of our codebase. 

Independent of the test, I also manually verified that this works end to end and the error is printed from a non-tty terminal:
```
Uploading bundle files to /Users/shreyas.goenka@databricks.com/.bundle/bundle playground 3/development/files...
The following UC schemas will be deleted or recreated. Any underlying data may be lost:
  delete schema foo
Error: the deployment requires destructive actions, but current console does not support prompting. Please specify --auto-approve if you would like to skip prompts and proceed
```